cpp/src/arrow/adapters/orc/adapter.cc

```diff
@@ -439,8 +439,28 @@ class ORCFileReader::Impl {
     return Status::OK();
   }
 
+  Status NextBatchReader(int64_t batch_size, const std::vector<std::string>& include_names,
+                          std::shared_ptr<RecordBatchReader>* out) {
```

Comment:
Can you refactor this so as to share most code with the other method above?

Reply:
Followed the "Status NextStripeReader() "  implementation.